### ISSUE TITLE
Consolidate ConflictsWith errors

### ISSUE THREAD

**Issue by danawillow** [hashicorp/terraform-plugin-sdk#286]
### SDK version

```
v1.4.0
```

### Use-cases

Have a single error message for two related ConflictsWith fields- see https://github.com/terraform-providers/terraform-provider-google/issues/5284.

### Attempted Solutions

If two fields conflict with each other, we can get a single error message by only setting `ConflictsWith` on one of them, which works fine.

However, if three fields (A, B, and C) conflict with each other, we would have to set `ConflictsWith` on at least two of them in order to cover all cases (setting it only on A would mean that a user could set B and C together in their config), which means that a user could end up with two error messages.

### Proposal

Consolidate error messages when possible. In the example in the related issue, instead of having both `Error: "name_prefix": conflicts with name` and `Error: "name": conflicts with name_prefix`, recognize that those two are the same and only return one of them. My guess is that we'd also want to do something similar to `ExactlyOneOf` and `AtLeastOneOf`.

### References

https://github.com/terraform-providers/terraform-provider-google/issues/5284

**Comment by jsoref** [comment]
For my reference, this appears to be: https://github.com/hashicorp/terraform-plugin-sdk/blob/97d4d4bd70641fb9ae7484c03c25b1f3d4383efe/helper/schema/schema.go#L1459

**Comment by radeksimko** [comment]
I agree with you @danawillow that these error messages and UX here could be improved.

The new "rich diagnostic errors/warnings" that came with 0.12 gave us the ability to carry helpful context alongside each message, so we can show the relevant snippet of code and point to the relevant line.

With that in mind we would need to tune this to allow us pointing to more than a single line of code, i.e. all conflicting fields, or we would need to arbitrarily pick one, which may confuse the user more than the current (verbose) messaging.

So aggregating the message itself is easy, aggregating the context is possible, but not as easy and we can't quite separate the two from each other as that is how errors & warnings are represented in the plugin protocol since 0.12.
